delivery_carrier_label_gls/stock.py

```diff
@@ -0,0 +1,337 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#  license AGPL version 3 or later
+#  see license in __openerp__.py or http://www.gnu.org/licenses/agpl-3.0.txt
+#  Copyright (C) 2014 Akretion (http://www.akretion.com).
+#  @author David BEAL <david.beal@akretion.com>
+#
+##############################################################################
+
+from openerp.osv import orm  # fields
+from openerp.tools.translate import _
+from .report.label import Gls, InvalidDataForMako
+from .report.exception_helper import (InvalidAccountNumber)
+from .report.label_helper import (
+    InvalidValueNotInList,
+    InvalidMissingField,
+    InvalidType,)
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from datetime import datetime
+import pycountry
+from operator import attrgetter
+
+
+EXCEPT_TITLE = "GLS Library Exception"
+LABEL_TYPE = 'zpl2'
+PACK_NUMBER = 0
+
+
+def raise_exception(orm, message):
+    raise orm.except_orm(EXCEPT_TITLE, map_except_message(message))
+
+
+def map_except_message(message):
+    """ Allows to map vocabulary from external library
+        to Odoo vocabulary in Exception message
+    """
+    model_mapping = {
+        'shipper_country': 'partner_id.country_id.code',
+    }
+    for key, val in model_mapping.items():
+        message = message.replace(key, val)
+    return message
+
+
+class StockPicking(orm.Model):
+    _inherit = "stock.picking"
+
+    def action_done(self, cr, uid, ids, context=None):
+        """
+        :return: see original method
+        """
+        if context is None:
+            context = {}
+        for picking in self.browse(cr, uid, ids, context=context):
+            if picking.carrier_type == 'gls':
+                self.generate_labels(
```

Comment:
I think it would be better to have an option to print on done. This could be interesting in base module as I don't think it is carrier specific.